routes/api/v1/favorites.js

```diff
@@ -18,4 +24,97 @@ router.post('/', (request, response) => {
     });
 });
 
+router.get('/', (request, response) => {
+  // console.log(request.body.api_key)
+  database('users').where('api_key', request.body.api_key)
+
+////////////////////////////////////////////////////////////////////////////////
+  .then(user => {
+    // console.log(user[0].id)
+    return database('favorites').where( 'user_id', user[0].id)
+  })
+
+////////////////////////////////////////////////////////////////////////////////
+  .then((favorites) => {
+    favorites_array = favorites.map(function (favorite_object) {
+    console.log(favorite_object.city)
+    return favorite_object.city
+});
+console.log(favorites_array)
+return(favorites_array)
+})
+
+////////////////////////////////////////////////////////////////////////////////
+.then((array) => {
+array.forEach((city) => {
+  var options = {
+      uri: 'https://maps.googleapis.com/maps/api/geocode/json',
+      qs: {
+          key: process.env.GOOGLE_API_KEY,
+          address: city
+      },
+      json: true
+  };
+
+rp(options).then(body => {
+    latitude = body.results[0].geometry.location.lat;
+    longitude = body.results[0].geometry.location.lng;
+    return body.results[0].geometry.location
+})
+
+
+////////////////////////////////////////////////////////////////////////////////
+.then((whatIsThis) => {
+    // officialResponse = "lalaal";
+    var options2 = {
+        uri: `https://api.darksky.net/forecast/${process.env.DARK_SKY_API_KEY}/${latitude},${longitude}?exclude=minutely`,
+        json: true
+    };
+
+    rp(options2)
+
+////////////////////////////////////////////////////////////////////////////////
+    .then(body2 => {
+      var weatherHash = {}
+      weatherHash["location"] = city;
+      weatherHash["currently"] = body2.currently
+
+      delete weatherHash.currently.time
+      delete weatherHash.currently.nearestStormDistance
+      delete weatherHash.currently.nearestStormBearing
+      delete weatherHash.currently.apparentTemperature
+      delete weatherHash.currently.dewPoint
+      delete weatherHash.currently.uvIndex
+      delete weatherHash.currently.ozone
```

Comment:
This is pretty cumbersome. I think it would be cleaner to create a class that only pulls out the data that you want. Then you could do something like:
```
var weather = new cityWeather(body2)
jsonResponse.push(weather)
```

Reply:
Thanks for the input. I did move that code to a model so that it is not in the route anymore.

However, I kept it to where some code is being deleted from the incoming JSON response from Dark Sky Api rather than creating a whole json response in a POJO. I think it is actually significantly less code to do it this way, and it looks cleaner to me.